pkg/api/api.go

```diff
@@ -53,17 +67,124 @@ func GetLicense(c *gin.Context) {
 			Status:    http.StatusBadRequest,
 			Message:   fmt.Sprintf("no license with shortname '%s' exists", queryParam),
 			Error:     err.Error(),
-			Path:      fmt.Sprintf("/api/license/%s", queryParam),
+			Path:      c.Request.URL.Path,
 			Timestamp: time.Now().Format(time.RFC3339),
 		}
 		c.JSON(http.StatusBadRequest, er)
 		return
 	}
 
 	res := models.LicenseResponse{
-		Data:   []models.License{license},
+		Data:   []models.LicenseDB{license},
 		Status: http.StatusOK,
+		Meta: models.Meta{
+			ResourceCount: 1,
+		},
 	}
 
 	c.JSON(http.StatusOK, res)
 }
+
+func CreateLicense(c *gin.Context) {
+	var input models.LicenseInput
+
+	if err := c.ShouldBindJSON(&input); err != nil {
+		er := models.LicenseError{
+			Status:    http.StatusBadRequest,
+			Message:   "invalid json body",
+			Error:     err.Error(),
+			Path:      c.Request.URL.Path,
+			Timestamp: time.Now().Format(time.RFC3339),
+		}
+		c.JSON(http.StatusBadRequest, er)
+		return
+	}
+
+	if input.Active == "" {
+		input.Active = "t"
+	}
+	license := models.LicenseDB(input)
+
+	result := DB.FirstOrCreate(&license)
+	if result.RowsAffected == 0 {
+
+		er := models.LicenseError{
+			Status:    http.StatusBadRequest,
+			Message:   "can not create license with same shortname",
+			Error:     fmt.Sprintf("Error: License with shortname '%s' already exists", input.Shortname),
+			Path:      c.Request.URL.Path,
+			Timestamp: time.Now().Format(time.RFC3339),
+		}
+		c.JSON(http.StatusBadRequest, er)
+		return
+	}
+	if result.Error != nil {
+		er := models.LicenseError{
+			Status:    http.StatusInternalServerError,
+			Message:   "Failed to create license",
+			Error:     result.Error.Error(),
+			Path:      c.Request.URL.Path,
+			Timestamp: time.Now().Format(time.RFC3339),
+		}
+		c.JSON(http.StatusInternalServerError, er)
+		return
+	}
+	res := models.LicenseResponse{
+		Data:   []models.LicenseDB{license},
+		Status: http.StatusCreated,
+		Meta: models.Meta{
+			ResourceCount: 1,
+		},
+	}
+
+	c.JSON(http.StatusCreated, res)
+}
+
+func UpdateLicense(c *gin.Context) {
+	var update models.LicenseDB
+	var license models.LicenseDB
+	shortname := c.Param("shortname")
+	if err := DB.First(&license, shortname).Error; err != nil {
```

Comment:
```suggestion
	if err := DB.Where("shortname = ?", shortname).First(&license).Error; err != nil {
```
Either you can provide condition in First(val, cond) or put a where before.